lucupy/minimodel/observation.py

```diff
@@ -262,6 +264,13 @@ def part_time(self) -> timedelta:
         """
         return sum((atom.part_time for atom in self.sequence), start=ZeroTime)
 
+    def cumulative_exec_times(self) -> List[timedelta]:
```

Comment:
I understand what this function is doing, but why are we switching back and forth from `timedelta` to `total_seconds` and then to `numpy`?  We could just do:
```python
cum_seq = [atom.exec_time if not atom.observed else timedelta() for atom in self.sequence]
return np.cumsum(cum_seq)
```
and return an `npt.NDArray[timedelta]`, and then, if necessary, convert and entry in it to seconds (`timedelta` is almost always stored in seconds unless the granularity is as high as days).

I guess I'm confused as to when and where we will be using this as well. Is it in GM? Could you add a comment as to why we would need the cumulative sum of the execution times of the unobserved atoms?

Reply:
Is used in GM with `timedelta` and np.cumsum only accepts numbers.

Reply:
I tried on my local machine and `np.cumsum` worked fine with a `List[timedelta]`. It's not a big deal either way. Just seems cleaner.

```python
>>> from datetime import timedelta
>>> my_list = [timedelta(seconds=3), timedelta(minutes=4), timedelta(hours=5)]
>>> np.cumsum(my_list)
array([datetime.timedelta(seconds=3), datetime.timedelta(seconds=243),
       datetime.timedelta(seconds=18243)], dtype=object)
```

Reply:
Uh, I try it and it didn't work. I'll add it thanks!